WordPress/openverse
# Combine related media component using `VMediaCollection.vue`

## Fixes


Extracted from #3785 to make it easier to review.
Fixes #3832 by @obulat

_Note: 428 lines of the 987 added lines are from "tapes" (saved API responses used for tests)_

## Description


We currently have two separate components for related images and related audio that are basically duplicating each other. This PR combines the 2 related components.

A bigger change is the new `VMediaCollection` component (originally introduced in #3785, but split from it to make reviewing simpler).

This component handles displaying the header, grid skeleton if fetching or the media collection if fetched, and the footer with load more button and the external search form, if necessary.

To simplify this component, I used slots for header and footer, and created `wrapper` components (similar to the audio templates) for the related, search and collection results. This PR contains the `VMediaCollection` and the `VRelatedMedia.vue` wrapper component. It does not use footer because the related media does not have load more button (or the external search form).

This PR renames the "old" components  for audio and image grid to minimize the number of changes (`VImageGrid` and `VAudioList` are "old", and `VImageCollection` and `VAudioCollection` are the new ones).

There's also a tiny unrelated change: the loading skeleton for image single result is now centered (using `mx-auto` class).

## Testing Instructions

Select an audio result and click on one of the related audios. You should see the `SELECT_RESULT` event with the `relatedTo` set correctly. The related images/audio should look correct.

Look at the single result pages. The related media section should look and behave correctly.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
